diff --git a/include/target.mk b/include/target.mk
index d13902a..265494e 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -18,8 +18,11 @@ DEFAULT_PACKAGES:=\
 	fstools \
 	libc \
 	libgcc \
-	libustream-mbedtls \
+	libustream-openssl \
 	logd \
+	luci \
+	luci-mod-admin-full \
+	luci-theme-bootstrap \
 	mtd \
 	netifd \
 	uci \
@@ -233,7 +236,7 @@ LINUX_RECONF_DIFF = $(SCRIPT_DIR)/kconfig.pl - '>' $(call __linux_confcmd,$(filt
 ifeq ($(DUMP),1)
   BuildTarget=$(BuildTargets/DumpCurrent)
 
-  CPU_CFLAGS = -Os -pipe
+  CPU_CFLAGS = -O2 -pipe
   ifneq ($(findstring mips,$(ARCH)),)
     ifneq ($(findstring mips64,$(ARCH)),)
       CPU_TYPE ?= mips64
@@ -277,7 +280,7 @@ ifeq ($(DUMP),1)
   ifeq ($(ARCH),aarch64)
     CPU_TYPE ?= generic
     CPU_CFLAGS_generic = -mcpu=generic
-    CPU_CFLAGS_cortex-a53 = -mcpu=cortex-a53
+    CPU_CFLAGS_cortex-a53 = -mcpu=cortex-a53+crypto+simd+crc
   endif
   ifeq ($(ARCH),arc)
     CPU_TYPE ?= arc700
diff --git a/config/Config-devel.in b/config/Config-devel.in
index cbac91c..7506bf7 100644
--- a/config/Config-devel.in
+++ b/config/Config-devel.in
@@ -153,7 +153,7 @@ menuconfig DEVEL
 
 	config EXTRA_OPTIMIZATION
 		string "Additional compiler options" if DEVEL
-		default "-fno-caller-saves -fno-plt" if !CONFIG_EXTERNAL_TOOLCHAIN && !arc
+		default "-fno-caller-saves -fno-plt -fomit-frame-pointer" if !CONFIG_EXTERNAL_TOOLCHAIN && !arc
 		default "-fno-caller-saves"
 		help
 		  Extra target-independent optimizations to use when building for the target.
diff --git a/config/Config-build.in b/config/Config-build.in
index 02931a8..de5bb57 100644
--- a/config/Config-build.in
+++ b/config/Config-build.in
@@ -46,6 +46,7 @@ menu "Global build settings"
 		bool "Select all userspace packages by default"
 		select ALL_KMODS
 		select ALL_NONSHARED
+		default n
 
 	config BUILDBOT
 		bool "Set build defaults for automatic builds (e.g. via buildbot)"
@@ -168,6 +169,7 @@ menu "Global build settings"
 	config USE_LTO
 		bool
 		prompt "Use the link-time optimizer for all packages (EXPERIMENTAL)"
+		default y
 		help
 		  Adds LTO flags to the CFLAGS and LDFLAGS.
 		  Packages can choose to opt-out via setting PKG_BUILD_FLAGS:=no-lto
